packages/analyticsdx-template-lint/src/linter.ts

```diff
@@ -376,6 +376,38 @@ export abstract class TemplateLinter<
     });
   }
 
+  private async lintVariablesNamesInObject(
+    templateInfo: JsonNode,
+    doc: Document,
+    valuesNode: JsonNode | undefined,
+    // errorCode should one of the values in ERRORS
+    errorCode: typeof ERRORS[keyof typeof ERRORS]
+  ) {
+    if (valuesNode?.type === 'object' && valuesNode.children && valuesNode.children.length > 0) {
+      const variableTypes = (await this.loadVariableTypesForTemplate(templateInfo)) || {};
+      const fuzzySearch = fuzzySearcher({
+        // make an iterable, to lazily call Object.keys() on if fuzzySearch is called
+        [Symbol.iterator]: () => Object.keys(variableTypes).filter(isValidVariableName)[Symbol.iterator]()
```

Comment:
Is the idea here avoiding doing `Object.keys(variableTypes).filter(isValidVariableName)` unless it is needed for searching (line 398)?

Reply:
Yep.